### ISSUE TITLE
[REQUEST] please add filters from trakt

### ISSUE THREAD

**Issue by jsox79** [nixgates/plugin.video.seren#866]
please add the filter option in trakt.  so we can use our filters vs cumbersome lists
thanks

**Comment by nixgates** [comment]
I've been wanting to add the ability to set custom filters for a while now.

Each time I've sat down to think about it, I haven't found a way that felt pleasant to use.

If you have any ideas on how you would like it implemented, I'm all ears!

**Comment by jsox79** [comment]
yes it should be possible via the api, ill shoot you a donation for your time.
it only works with vip trakt but i think its worth it, since seren already pretty much a premium app
heres the documentation.  i completely missed it a trakt dev pointed it out to me.
so you could do it a my saved filters section or something.
https://trakt.docs.apiary.io/#reference/users/approve-or-deny-follower-requests/get-saved-filters
you flesh it out man and send me a donate link.  thanks

**Comment by npmstart-pray** [comment]
@nixgates asked for your input, @jsox79 - not an assignment. 

Perhaps you can suggest how to make it work for those of you who are trakt VIP without negatively affecting how Seren works for non-VIP users...? Or do you know how to make the trakt filtering work for everyone, VIP or not?

**Comment by jsox79** [comment]
@npmstart-pray this was a feature request, who does feature requests?  the person responsible for the app development.  i dont think you understand how this works.  @nixgates doesn't require you to ride in like a white knight to save em from the evil guy making feature requests in the feature request section.  he asked how to implement it, i can gather he is already familiar with the trakt api calls, i showed him the documentation as i do not have a unique knowledge of the underpinnings of seren.

regarding filters for vip trakt, its a feature that is only available for trakt vip users, so youd need to contact trakt for that.

**Comment by npmstart-pray** [comment]
"Figure it out and I'll pay you" was the gist of your post. Nix was asking for your input, not for a donation/funding- give them a more specific direction to try other than "make it work" because they imply there are a number of possibilities. Which of them would you prefer they explore first? That's the discussion required. This is a waste of time.

**Comment by jsox79** [comment]
man i was offering to pay him for his services which are usually free.  you are totally trolling.  eventually lists are going to go away on trakt and filters will be all thats available according to their devs.  lists bog down their api.  so it should be on the roadmap.  i think you totally missed the context of my request.  hes already using the api to make calls for lists, trending, popular etc.  adding in what i asked would be practically the same.  if it came off as douchey it was not my intent.  many people come with demands and never offer anything in return.  i was merely offering to pay the them for their services on a feature that would benefit many.  im sorry that triggered you in some weird way, but you need to stop hijacking other peoples threads.  you have nothing to contribute but angst and some perceived slight to nix like your a guardian angel.  this isnt reddit or a forum. its a feature request on github, your opinion isnt really needed, nor does it help.  go be a hall monitor somewhere else karen.

**Comment by drinfernoo** [comment]
@jsox79 @npmstart-pray Let's keep it civil, please.

There is no problem with requesting features, but @nixgates will have to comment on whether he was offended by the implication of donation for time spent.

@jsox79 Thanks for the link to the documentation. That will be handy if we ever get there.

> who does feature requests? the person responsible for the app development.

Feature requests on this **public** and **open-source** repository are welcome (and encouraged), as are bug reports, obviously. As this is an open-source project, pull requests are also viewed positively, giving anyone who feels comfortable the opportunity to contribute. The bulk of in-development features and fixes do happen in a private repository, but public pull requests are always reviewed, and often incorporated before a public release.

More to the point (and let's attempt to keep it that way), I think the main sticking point with implementing this is not necessarily one of API access, but one of user experience. How should the process of using filters look while inside of Kodi? How does one change filtering options? Will this need some sort of custom window? Those are just a few of the questions at hand.

**Comment by jsox79** [comment]
@drinfernoo i too want to be civil, i was just lambasted by some bystander because he felt I offended nix, which again was totally not my intention.  i was only offering to pay for the feature add.  i shouldnt of added the karen/hall monitor jab at the end, but it felt appropriate at the time.  

as far as filters they happen on the trakt side, so in the trakt app i can create filters and save them, these filters can be recalled much like lists are now.  so when you do an api call for saved filters in the tv shows category, it will return your different filters.  so these filters are predefined in trakt, you're not actively creating the filters in kodi, however that would be neat, but a little cumbersome and clunky.  easier to just predefine your filters or like other filters other people have made.

im not familiar with which piece of code in seren handles those calls in the my movies/tv section.  point me in the direction and ill definitely take a stab at it.  if i can see how a call is handled in python i can use logic to reconstruct a similar call through trial an error.  and again my intent was never to insult anyone who works on this project.  thanks for your help.

**Comment by nixgates** [comment]
@npmstart-pray while I thank you for coming to my defence, I'm sure no harm or malice was intended in this instance.

The Dr is right, let's all keep a cool head here.

@jsox79 funnily enough! I didn't know Trakt had this feature! I definetely need to look more into the VIP stuff of Trakt! I originally thought the request was to do something similar to this Trakt feature but locally inside Seren. Allowing the user to build custom menus inside Seren! Hence the concern for how to implement it!

This endpoint shouldn't be to hard to implement seeing that Trakt provides all the required information we would need!

Also, no donation necessary!

**Comment by jsox79** [comment]
lol i didnt know it was there either, the dev was like its here.  but its so not in the right place, its nested under something else.  not a very good place for it in the documentation.  unless you read the whole thing from start to finish which lets be honest, we're only wanting the bits and pieces we need right lol.  thanks for your hard work on this project.  there are so many forks, seren is just the best imo.  and the speed increase is noticeable in python 3 update, been loving that.

**Comment by nixgates** [comment]
@jsox79 Yea I generally only rip through the exact API call I need to view 90% of the time lol.

If I could ask, give us a little bit. As we just pushed a recent major version our current focus is on bug fixes at this moment. Wont be long though I promise!

**Comment by jsox79** [comment]
no worries, take your time.  thanks again for your hard work.
